src/uipath/_cli/_evals/_progress_reporter.py

```diff
@@ -586,11 +593,18 @@ def _collect_coded_results(
             if eval_result.evaluator_id not in evaluators:
                 continue
 
+            # Convert BaseModel justification to JSON string for API compatibility
+            justification = eval_result.result.details
+            if justification is not None and hasattr(justification, "model_dump"):
+                justification = json.dumps(justification.model_dump())
+            elif justification is not None and not isinstance(justification, str):
+                justification = json.dumps(justification)
```

Comment:
This won't work for items that are not json serializable.

Reply:
Actually nvm. I just looked at the `details` type
```
details: Optional[str | BaseModel] = None
```

I would suggest rewriting this block as 
```
            if isinstance(eval_result.result.details, BaseModel):
                justification = json.dumps(justification.model_dump())
           elif isinstance(eval_result.result.details, str):
               justification = eval_result.result.details
           else:
              justification = None
```

Note: at this point, this is a nit.